pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -253,43 +253,46 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 		return fmt.Errorf("node %s doesn't have providerID", nodeName)
 	}
 
-	cidrs, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
+	cidrStrings, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
 	if err != nil {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: %v", err)
 	}
-	if len(cidrs) == 0 {
+	if len(cidrStrings) == 0 {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: Node %v has no CIDRs", node.Name)
 	}
-	_, cidr, err := net.ParseCIDR(cidrs[0])
+	if len(cidrStrings) > 2 {
+		//Can have at most 2 ips (one for v4 and one for v6)
+		klog.Infof("Got more than 2 ips: %v, truncating to 2", cidrStrings)
+		cidrStrings = cidrStrings[:2]
+	}
+
+	cidrs, err := netutils.ParseCIDRs(cidrStrings)
 	if err != nil {
-		return fmt.Errorf("failed to parse string '%s' as a CIDR: %v", cidrs[0], err)
+		return fmt.Errorf("failed to parse strings %v as CIDRs: %v", cidrStrings, err)
 	}
-	podCIDR := cidr.String()
+	podCIDRs := cidrsAsString(cidrs)
```

Comment:
isn't cidrsAsString the same as cidrStrings?

Reply:
I had wrong assumption that converting to net.IPNet and back to string with .String() would get us consistent IP strings.

Changed the code to handle comparisons with net.IPNet format and added a test case where 2001:db8::/64 and 2001:db8:0::/64 should be considered as same IPs